src/ServiceBus/ServiceBus.Autorest/custom/Set-AzServiceBusNamespaceV2.ps1

```diff
@@ -0,0 +1,238 @@
+            $hasDisableLocalAuth = $PSBoundParameters.Remove('DisableLocalAuth')
+            $hasKeyVaultProperty = $PSBoundParameters.Remove('KeyVaultProperty')
+            $hasUserAssignedIdentity = $PSBoundParameters.Remove('UserAssignedIdentity')
+            $hasIdentityType = $PSBoundParameters.Remove('IdentityType')
+            $hasMinimumTlsVersion = $PSBoundParameters.Remove('MinimumTlsVersion')
+            $hasRequireInfrastructureEncryption = $PSBoundParameters.Remove('RequireInfrastructureEncryption')
+            $hasPublicNetworkAccess = $PSBoundParameters.Remove('PublicNetworkAccess')
+            $hasSkuCapacity = $PSBoundParameters.Remove('SkuCapacity')
+            $hasTag = $PSBoundParameters.Remove('Tag')
+            $hasDefaultProfile = $PSBoundParameters.Remove('DefaultProfile')
+            $hasAsJob = $PSBoundParameters.Remove('AsJob')
+            $null = $PSBoundParameters.Remove('WhatIf')
+            $null = $PSBoundParameters.Remove('Confirm')
+            Write-Host @PSBoundParameters
+            $serviceBusNamespace = Get-AzServiceBusNamespaceV2 @PSBoundParameters
+
+            # 2. PUT
+            $null = $PSBoundParameters.Remove('InputObject')
+            $null = $PSBoundParameters.Remove('ResourceGroupName')
+            $null = $PSBoundParameters.Remove('NamespaceName')
+            $null = $PSBoundParameters.Remove('Name')
+            $null = $PSBoundParameters.Remove('SubscriptionId')
+
+            if ($hasAsJob) {
+                $PSBoundParameters.Add('AsJob', $true)
+            }
+
+            if ($hasAlternateName) {
+                $serviceBusNamespace.AlternateName = $AlternateName
+            }
+
+            if ($hasDisableLocalAuth) {
+                $serviceBusNamespace.DisableLocalAuth = $DisableLocalAuth
+            }
+            if ($hasKeyVaultProperty) {
+                $serviceBusNamespace.KeyVaultProperty = $KeyVaultProperty
+                $serviceBusNamespace.KeySource = 'Microsoft.KeyVault'
+            }
+            if ($hasIdentityType) {
+                $serviceBusNamespace.IdentityType = $IdentityType
+            }
+            if($hasRequireInfrastructureEncryption){
+                $serviceBusNamespace.RequireInfrastructureEncryption = $RequireInfrastructureEncryption
+            }
+            if ($hasUserAssignedIdentity) {
+                $serviceBusNamespace.UserAssignedIdentity = $UserAssignedIdentity
+            }
+            if ($hasMinimumTlsVersion) {
+                $serviceBusNamespace.MinimumTlsVersion = $MinimumTlsVersion
+            }
+            if ($hasPublicNetworkAccess) {
+                $serviceBusNamespace.PublicNetworkAccess = $PublicNetworkAccess
+            }
+            if ($hasSkuCapacity) {
+                $serviceBusNamespace.SkuCapacity = $SkuCapacity
+            }
+            if ($hasTag) {
+                $serviceBusNamespace.Tag = $Tag
+            }
+            if ($hasDefaultProfile) {
```

Comment:
DefaultProfile should not be handled like this. Please check my EventHubs PR for how to handle it.